src/client.cpp

```diff
@@ -50,6 +50,7 @@ CHelper_libXBMC_addon *XBMC      = NULL;
 CHelper_libXBMC_pvr   *PVR       = NULL;
 PVR_MENUHOOK          *menuHook  = NULL;
 CTvheadend            *tvh       = NULL;
+tvheadend::Settings              *settings  = NULL;
```

Comment:
I thought you were going to eliminate the settings singleton? Instead you are introducing another one here? This does not make sense at all, sorry.

Reply:
the addon stays a singleton as long those globals are here. there is work in progress to change this. addon like inputstream are already changed to new design.